core/src/main/java/edu/wpi/grip/core/operations/networktables/NTPublishOperation.java

```diff
@@ -0,0 +1,113 @@
+package edu.wpi.grip.core.operations.networktables;
+
+import com.google.common.eventbus.EventBus;
+import edu.wpi.first.wpilibj.networktables.NetworkTable;
+import edu.wpi.first.wpilibj.tables.ITable;
+import edu.wpi.grip.core.*;
+
+import java.io.InputStream;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * An operation that publishes any type that implements {@link NTPublishable} to NetworkTables.
+ * <p>
+ * To be publishable, a type should have one or more accessor methods annotated with {@link NTValue}.  This is done
+ * with annotations instead of methods
+ */
+public class NTPublishOperation<T extends NTPublishable> implements Operation {
+
+    private final NetworkTable table;
+    private final Class<T> type;
+    private final List<Method> ntValueMethods = new ArrayList<>();
+
+    public NTPublishOperation(Class<T> type) {
+        NetworkTable.setClientMode();
+        NetworkTable.setIPAddress("localhost"); // TODO
+
+        this.table = NetworkTable.getTable("GRIP");
+        this.type = type;
+
+        // Any accessor method with an @NTValue annotation can be published to NetworkTables.
+        for (Method method : type.getDeclaredMethods())
+            if (method.getAnnotation(NTValue.class) != null)
+                ntValueMethods.add(method);
```

Comment:
Check that all methods annotated don't take arguments. This will just cause problems later.